### PR TITLE
[MDX] Expose `jsToMdxTreeNode` util

### PR DESCRIPTION
**Pull request by bholmesdev**
## Changes

We've had some questions about an `injectScript('page-ssr')` equivalent for MDX. Today, `injectScript('page-ssr')`, _will_ work for Markdown and MDX for injecting styles and scripts that run server-side, but can't be used for the same trickery you can achieve in `.astro` files. Namely, you can't inject component imports for use in the file itself.

This exposes a utility for MDX plugin authors to achieve the same effect: `jsToMdxTreeNode` ([see usage example](https://github.com/withastro/astro/pull/4890/files#diff-34e236c72f3f9cc4c15d21cc79395f13b57f4a63e9e3a775fd9cf5450452c252R9-R22)). This should make JS injection in MDX a bit easier to document, since MDX's documentation for their custom tree nodes is basically nonexistent.

## Testing

Test that injecting a `components` export works as expected.

## Docs

TODO: update README if we agree this is a good idea!

### CODE REVIEW

File: packages/integrations/mdx/src/internal-utils.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+ * Convert a string of JavaScript to an MDX-compatible tree node
+ * for use in remark or rehype plugins
+ * @param jsString String of valid JavaScript
+ * @param acornOpts Acorn parser options ([see Acorn docs](https://github.com/acornjs/acorn/tree/master/acorn#interface) for more).
+ * @returns An MDX-compatible tree node
+ */
+export function jsToMdxTreeNode(
+	jsString: string,
+	acornOpts: AcornOpts = {
+		ecmaVersion: 'latest',
```
Comment (christian-hackyourshack) [inline packages/integrations/mdx/src/internal-utils.ts:75]: If someone just wants to change the ecmaVersion, the sourceType will be lost, better set the defaults individually.

File: packages/integrations/mdx/src/internal-utils.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		sourceType: 'module',
+	}
+): MdxjsEsm {
+	return {
+		type: 'mdxjsEsm',
+		value: '',
```
Comment (christian-hackyourshack) [inline packages/integrations/mdx/src/internal-utils.ts:81]: This should be your jsString. Not strictly required, but improves debuggability.

**Review by christian-hackyourshack — COMMENTED** [review]
My general thought on this is: This is really nice to have, but it is not really necessary, because it does not unlock anything, that you could not easily do outside Astro. On the other hand you add something to the public API that needs to be maintained for a long time (you can hear, I am getting old).

It also does not really help plugin authors save duplicate work, like eg. parsing the frontmatter to be able to access it. Hence, I would suggest to keep it internal and maybe we can write a small library astro-mdx-utils (actually it's probably a general remark-mdx-utils library) that can be evolved independent from the mdx integration.

**Comment by bholmesdev** [comment]
Closing after discussing the right abstraction for this! We'll keep sugar like this to framework authors and community packages like m2dx.

**Pull request closed without merging**